tests/build.gradle

```diff
@@ -66,7 +66,8 @@ dependencies {
     compile 'org.scalamock:scalamock-scalatest-support_2.11:3.4.2'
     compile 'com.typesafe.akka:akka-testkit_2.11:2.4.16'
     compile 'com.typesafe.akka:akka-http-testkit_2.11:10.0.10'
-    compile 'com.github.java-json-tools:json-schema-validator:2.2.8';
+    compile 'com.github.java-json-tools:json-schema-validator:2.2.8'
+    compile "org.mockito:mockito-core:2.15.0"
```

Comment:
We already have `scalamock` in the repo, can you use that?

Reply:
Checked scalamock but did not found any way to [spy][1] on existing object which I require in the test. 

Scalamock can mock any interface or class but looks like does not provide a way to proxy an existing object and check interactions that happen with that object

[1]: http://static.javadoc.io/org.mockito/mockito-core/2.15.0/org/mockito/Mockito.html#13

Reply:
You might be right, in this case you'd need to "implement" the method of the mock.